### DISCUSSION TITLE
Specifying a persistent volume claim for queue

### DISCUSSION THREAD

**Discussion by mattkallo** [arkmq-org/arkmq-org-broker-operator#509] [Q&A] [answered]
Hi - I am using EKS k8s cluster and have a persistent volume claim (EFS) to which we want to persist all files related to the queue. I configured persistenceEnabled to true, but how do I set the volume? Could you pls give me some pointers?

**Comment by brusdev** [comment, marked as answer]
Hi @mattkallo, each broker instance needs its own PV. By default, when you set `spec.deploymentPlan.persistenceEnabled` to true, a new PVC is created for each broker instance and you can customize the size and the storageClassName, using the following properties: `spec.deploymentPlan.storage.size` and `spec.deploymentPlan.storage.storageClassName`.
